docs/core/whats-new/dotnet-7.md

```diff
@@ -0,0 +1,181 @@
+| Attribute to indicate what what kind of syntax is expected in a string | <xref:System.Diagnostics.CodeAnalysis.StringSyntaxAttribute?displayProperty=fullName> | For example, you can specify that a `string` parameter expects a regular expression by attributing the parameter with `[StringSyntax(StringSyntaxAttribute.Regex)]`. |
+
+## Observability
+
+.NET 7 makes improvements to *observability*. Observability helps you understand the state of your app as it scales and as the technical complexity increases. .NET's observability implementation is primarily built around [OpenTelemetry](https://opentelemetry.io/). Improvements include:
+
+- The new <xref:System.Diagnostics.Activity.CurrentChanged?displayProperty=nameWithType> event, which you can use to detect when the span context of a managed thread changes.
+- New, performant enumerator methods for <xref:System.Diagnostics.Activity> properties: <xref:System.Diagnostics.Activity.EnumerateTagObjects>, <xref:System.Diagnostics.Activity.EnumerateLinks>, and <xref:System.Diagnostics.Activity.EnumerateEvents>.
+
+For more information, see the [.NET 7 Preview 4](https://devblogs.microsoft.com/dotnet/announcing-dotnet-7-preview-4/#observability) blog post.
+
+## .NET SDK
+
+The .NET 7 SDK improves the CLI template experience. It also enables publishing to containers and central package management with NuGet.
+
+### Templates
+
+Some welcome improvements have been made to the `dotnet new` command and to template authoring.
+
+#### dotnet new
+
+The [`dotnet new`](../tools/dotnet-new.md) CLI command, which creates a new project, configuration file, or solution based on a template, now supports [tab completion](../tools/enable-tab-autocomplete.md) for exploring:
+
+- Available template names
+- Template options
+- Allowable option values
+
+In addition, for better conformity, the `install`, `uninstall`, `search`, `list`, and `update` subcommands no longer have the `--` prefix.
+
+#### Authoring
+
+Template *constraints*, a new concept for .NET 7, let you define the context in which your templates are allowed. Constraints help the template engine determine which templates it should show in commands like `dotnet new list`. You can constrain your template to an operating system, a template engine host (for example, the .NET CLI or New Project dialog in Visual Studio), and an installed workload. You define constraints in your template's configuration file.
```

Comment:
```suggestion
Template *constraints*, a new concept for .NET 7, lets you define the context in which your templates are allowed. Constraints help the template engine determine which templates it should show in commands like `dotnet new list`. You can constrain your template to an operating system, a template engine host (for example, the .NET CLI or New Project dialog in Visual Studio), and an installed workload. You define constraints in your template's configuration file.
```

Reply:
I'm reading it as "template contstraints...let you define".